src/wp-includes/rest-api/endpoints/class-wp-rest-font-faces-controller.php

```diff
@@ -856,15 +856,18 @@ protected function sanitize_src( $value ) {
 	 */
 	protected function handle_font_file_upload( $file ) {
 		add_filter( 'upload_mimes', array( 'WP_Font_Utils', 'get_allowed_font_mime_types' ) );
-		// Filter the upload directory to return the fonts directory.
-		add_filter( 'upload_dir', '_wp_filter_font_directory' );
```

Comment:
This is the back-compat break I am concerned about. If this approach was desired an enhancement should have been opened in June, 2023 https://github.com/WordPress/gutenberg/pull/52704#discussion_r1269940796

Reply:
Thinking we have to "agree to disagree" here :) 
Don't think this is a back-compat problem but rather a fix to a bug that allows plugins to "behave badly" and disable part of the WP plugins API. Another good reason to fix this is the fact that there are no known plugins that do this "bad behavior".